ext/native/thin3d/VulkanQueueRunner.cpp

```diff
@@ -778,11 +786,45 @@ void VulkanQueueRunner::PerformRenderPass(const VKRStep &step, VkCommandBuffer c
 			break;
 
 		case VKRRenderCommand::BIND_PIPELINE:
-			if (c.pipeline.pipeline != lastPipeline) {
+			if (c.pipeline.pipeline != lastGraphicsPipeline) {
 				vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, c.pipeline.pipeline);
-				lastPipeline = c.pipeline.pipeline;
+				lastGraphicsPipeline = c.pipeline.pipeline;
+			}
+			break;
+
+		case VKRRenderCommand::BIND_GRAPHICS_PIPELINE:
+		{
+			VKRGraphicsPipeline *pipeline = c.graphics_pipeline.pipeline;
+			if (!pipeline->pipeline) {
```

Comment:
I wonder if there's any possibility of a race here - if it gets done and notifies *just* before we lock and do the wait below - but on the other hand, if it's done, it will never enter the while loop anyway. Hm. It probably is safe...

Reply:
Yeah, I think that sounds safe too.

-[Unknown]